qir/qat/Passes/ValidationPass/ValidationPass.cpp

```diff
@@ -295,7 +295,7 @@ bool ValidationPass::satisfyingExternalCallRequirements()
 {
     auto ret = true;
 
-    if (profile_config_.allowlistExternalCalls())
+    if (profile_config_.allowlistExternalCalls() && !qis_config_.allowAnyQis())
```

Comment:
I believe this code does not really distinguish between generic external calls and QIS calls.

Perhaps having two distinct `if` statements, one using `allowlistExternalCalls` as conditional which internally checks against `allowedExternalCallNames` for non-QIS functions, and another using `!allowAnyQis` as conditional which internally checks against `allowedQis` for QIS functions. @swernli what do you think?

Reply:
I think you make a great point. I adjusted the check to instead only allow functions that start with `__quantum__qis__` when the option is present and verified that other functions still get rejected. This makes the behavior line up with the parameter name more clearly.

Reply:
Thanks for this fix!